src/Stratis.Bitcoin/Base/TimeSyncBehavior.cs

```diff
@@ -273,16 +273,12 @@ private void RecalculateTimeOffsetLocked()
             if (unweightedSamplesCount >= MinUnweightedSamplesCount)
             {
                 this.logger.LogTrace("We have {0} unweighted samples, {1} outbound samples and {2} inbound samples.", unweightedSamplesCount, this.outboundTimestampOffsets.Count, this.inboundSampleSources.Count);
-                List<double> allSamples = this.inboundTimestampOffsets.Select(s => s.TimeOffset.TotalSeconds).ToList();
-
-                List<double> outboundOffsets = this.outboundTimestampOffsets.Select(s => s.TimeOffset.TotalSeconds).ToList();
-                for (int i = 0; i < OutboundToInboundWeightRatio; i++)
-                    allSamples.AddRange(outboundOffsets);
+                
+                var medianOffsetInSeconds = this.GetMedianOffsetInSecondsFromInboundAndWeightedOutbound();
```

Comment:
unvar

Reply:
Yes, I'm okay with more expressive name of variable for median, And I am also okay with separation of the logic to its own method (although I don't think that `GetMedianOffsetInSecondsFromInboundAndWeightedOutbound` is a useful name of such a method), but you can't just remove the type as discussed elsewhere. Please also see a comment about subjective improvements.